### PR TITLE
Source Mailchimp: fix the way request params are built

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/oncall/issues/1058

## How
- Do not use state as a source for request parameters. Use stream slices instead
- Set test strictness level to high

### CODE REVIEW

File: airbyte-integrations/connectors/source-mailchimp/integration_tests/expected_records.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...ST"}, {"rel": "feedback", "href": "https://us2.api.mailchimp.com/3.0/campaigns/49d68626f3/feedback", "method": "GET", "targetSchema": "https://us2.api.mailchimp.com/schema/3.0/Definitions/Campaigns/Feedback/CollectionResponse.json"}, {"rel": "content", "href": "https://us2.api.mailchimp.com/3.0/campaigns/49d68626f3/content", "method": "GET", "targetSchema": "https://us2.api.mailchimp.com/schema/3.0/Definitions/Campaigns/Content/Response.json"}, {"rel": "send_checklist", "href": "https://us2.api.mailchimp.com/3.0/campaigns/49d68626f3/send-checklist", "method": "GET", "targetSchema": "https://us2.api.mailchimp.com/schema/3.0/Definitions/Campaigns/Checklist/Response.json"}, {"rel": "pause", "href": "https://us2.api.mailchimp.com/3.0/campaigns/49d68626f3/actions/pause", "method": "POST"}, {"rel": "resume", "href": "https://us2.api.mailchimp.com/3.0/campaigns/49d68626f3/actions/resume", "method": "POST"}, {"rel": "replicate", "href": "https://us2.api.mailchimp.com/3.0/campaigns/49d68626f3/actions/replicate", "method": "POST"}, {"rel": "create_resend", "href": "https://us2.api.mailchimp.com/3.0/campaigns/49d68626f3/actions/create-resend", "method": "POST"}]}, "emitted_at": 1671548438209}
```
Comment (alafanechere) [inline airbyte-integrations/connectors/source-mailchimp/integration_tests/expected_records.txt:1]: This is likely disclosing internal mailchimp data. Please:
- erase it
- log an issue to set up a sandbox Mailchimp account
- bypass expected_records test

**Review by alafanechere — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-mailchimp/integration_tests/expected_records.txt (hunk 1 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-mailchimp/integration_tests/expected_records.txt]: thanks for flagging. done

Comment (ycherniaiev) [reply airbyte-integrations/connectors/source-mailchimp/integration_tests/expected_records.txt]: @alafanechere we have testing account.Just want to figure out what internal data are we disclosing.

Comment (alafanechere) [reply airbyte-integrations/connectors/source-mailchimp/integration_tests/expected_records.txt]: The data sample looks like past emailing campaigns results with email addresses we don't want to disclose.

**Comment by evantahler** [comment]
@davydov-d I'm going to delete this branch and close this PR.  I'll be asking github to scrub the history of this PR and branch as well to remove `expected_records` from being publicly available.  

Please re-create this PR, but squash your commits to only have the latest version (without expected_records)

**Comment by evantahler** [comment]
@davydov-d Thanks for making https://github.com/airbytehq/airbyte/issues/20726 to get a sandbox account.  In the mean time, I think it's OK to skip the tests which need expected_records to close the On Call issue.

**Comment by davydov-d** [comment]
@evantahler great, I'm on my way of re-creating the PR

**Pull request closed without merging**